docs/en/operation/Journal.md

```diff
@@ -88,8 +89,60 @@ $ bin/alluxio-start.sh -i <backup_uri> masters
 The `<backup_uri>` should be a full URI path that is available to all masters, e.g.
 `hdfs://[namenodeserver]:[namenodeport]/alluxio_backups/alluxio-journal-YYYY-MM-DD-timestamp.gz`
 
+If starting up masters individually, pass the `-i` argument to each one. The master which
+becomes primary first will import the journal backup, and the rest will ignore the `-i`.
+
 If the restore succeeds, you should see a log message along the lines of
 ```
 INFO AlluxioMasterProcess - Restored 57 entries from backup
 ```
 in the primary master logs.
+
+### Changing masters
+
+If the journal is stored in a shared storage system like HDFS, changing masters is easy.
+As long as the new master sets `alluxio.master.journal.folder` the same as the old
+master, it will start up in the same state that the old master left off.
+
+If the journal is stored on the local filesystem, the journal folder needs to be
+copied to the new master.
+
+## Advanced
+
+### Managing the journal size
+
+When running with a single master, the journal folder size will grow indefinitely
+as metadata operations are written to journal log files. To address this, production
+deployments should run in HA mode with multiple Alluxio masters. The non-primary
+masters will create checkpoints of the master state and clean up the logs that
+were written prior to the checkpoints. For example, if 3 million alluxio files were
+created and then 2 million were deleted, the journal logs would contain 5 million
+total entries. Then if a checkpoint is taken, the checkpoint will contain only the
+metadata for the 1 million remaining files, and the original 5 million entries will be deleted.
+
+By default, checkpoints are automatically taken every 2 million entries. This can be configured by
+setting `alluxio.master.journal.checkpoint.period.entries` on the masters. Setting
+the value lower will reduce the amount of disk space needed by the journal at the
+cost of additional work for the secondary masters.
+
+If your cluster has high metadata throughput, consider
+
+If HA mode is not an option, it is possible to run a master on the same node as a
+dedicated secondary master. The second master exists only to write checkpoints, and
+will not serve client requests if the primary master dies. In this setup, both
+masters have similar memory requirements since they both need to hold all Alluxio
+metadata in memory. To start a dedicated secondary master for writing periodic checkpoints,
+run
+
+```bash
+bin/alluxio-start.sh secondary_master
```

Comment:
nit: do we put `$` in the beginning of all bash commands?

Reply:
done, thanks